pkg/l2announcer/l2announcer.go

```diff
@@ -835,22 +835,18 @@ func (l2a *L2Announcer) processLeaderEvent(event leaderElectionEvent) error {
 }
 
 func (l2a *L2Announcer) recalculateL2EntriesTableEntries(ss *selectedService) error {
-	txn := l2a.params.StateDB.WriteTxn()
-	w := l2a.params.L2AnnounceTable.Writer(txn)
+	tbl := l2a.params.L2AnnounceTable
+	txn := l2a.params.StateDB.WriteTxn(tbl)
+	defer txn.Abort()
```

Comment:
IIUC abort will cause a metric update even in case a commit already happened (while the same metric is not updated on `Commit()`.

Reply:
:+1:  we'll follow up with metrics fixes after this PR.